theano/scan_module/scan_opt.py

```diff
@@ -351,14 +378,14 @@ def process_node(self, fgraph, node):
                 remove=[node],
                 reason='scanOp_pushout_nonseqs_ops')
             return True
-        elif to_keep == []:
+        elif not to_keep:
```

Comment:
I think that was recommended by pep8 or pylint(I don't remember which), that was why I changed it. I disagree with your argument for readability for me "not to_keep" is way more readable.